controllers/osbuild_controller_test.go

```diff
@@ -0,0 +1,422 @@
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+
+		})
+
+		It("Should return error when failed to get the instance", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(nil, errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultShortRequeue))
+		})
+	})
+
+	Context("Handle deletion", func() {
+		It("Should return Done if removed the finalizer successfully", func() {
+			// given
+			osbuildInstance.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+		})
+	})
+
+	Context("Create edge-container request ", func() {
+		It("and fail on postCompose with error", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, gomock.Any(), gomock.Any()).Return(nil)
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(nil, errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultLongRequeue))
+		})
+		It("and on postCompose returns status code bad request ", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, gomock.Any(), gomock.Any()).Return(nil)
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(&composerPostResponseFailed, nil)
+
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultLongRequeue))
+		})
+		It("and on postCompose returns status code created ", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, gomock.Any(), gomock.Any()).Return(nil)
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(&composerPostResponseCreated, nil)
+
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultLongRequeue))
+			Expect(osbuildInstance.Status.ContainerComposeId).To(Equal(composerPostResponseCreated.JSON201.Id.String()))
```

Comment:
Consider having a `Status` variable and then just compare it with the one from the `osbuildInstance` instead of multiple lines